pkg/db/transaction.go

```diff
@@ -46,27 +40,44 @@ func TransactionResult(conn Connection, dbOps func(tx *TxConnection) (interface{
 		if err != nil {
 			log("Rollback transactional DB context because an error occurred: %s", err)
 			if rollbackErr := txConnection.Rollback(); rollbackErr != nil {
-				err = errors.Wrap(err, fmt.Sprintf("Rollback of db operations failed: %s", rollbackErr))
+				log("Rollback failed: %s", rollbackErr)
+			}
+
+			if strings.Contains(err.Error(), "could not serialize access") {
+				log("Rollback caused by colliding transaction: will retry this transaction")
+				txErr = errors.Wrap(txErr, err.Error()) //remember this error when retrying
+				continue
 			}
+
 			return result, err
 		}
 
 		commitErr := txConnection.commit()
 		if commitErr == nil {
 			return result, nil
 		}
+		txErr = errors.Wrap(txErr, commitErr.Error()) //remember this error when retrying
 
-		txErr = errors.Wrap(txErr, commitErr.Error())
 		log("Rollback transactional DB context because commit failed: %s", commitErr.Error())
 		if txRollbackErr := txConnection.Rollback(); txRollbackErr != nil {
-			txErr = errors.Wrap(txErr, txRollbackErr.Error())
+			log("Rollback failed: %s", txRollbackErr)
 		}
 
 	}
 
 	return result, txErr
 }
 
+func randomJitter() time.Duration {
```

Comment:
is it make sense change to this? the jitter can be just a pure int number, right? If `* time.Millisecond` at the end.
```
	jitter := rand.Intn(txMaxJitter)
	if jitter < txMinJitter {
		jitter = jitter + txMinJitter
	}
	return time.Duration(jitter) * time.Millisecond
```

Reply:
The jitter should be dynamic to achieve some load balancing on DB side (right now, we can assume that the problem happens because too many transactions are trying to update their data at the same time).

Using a int-number which is not dynamic would lead to a next round of collisison when it retries all transactions at the same time later.

Reply:
Ok, I got the seed purpose, but then we just need to bring the seed in the beginning to offer a wide range of int numbers, but regarding the `jitter`, it's the same logic, just the time when it converts to `time.Duration`